storage/client_raft_test.go

```diff
@@ -239,7 +241,9 @@ func TestFailedReplicaChange(t *testing.T) {
 	}()
 	storage.TestingCommandFilter = func(method string, args proto.Request, reply proto.Response) bool {
 		if method == proto.EndTransaction && args.(*proto.EndTransactionRequest).Commit == true {
-			reply.Header().SetGoError(util.Errorf("boom"))
+			oldErr := unsafe.Pointer(&reply.Header().Error)
+			newErr := proto.Error{Message: "boom"}
+			atomic.SwapPointer((*unsafe.Pointer)(oldErr), unsafe.Pointer(&newErr))
```

Comment:
Slightly odd that you're casting `oldErr` (type `unsafe.Pointer`) to the type `*unsafe.Pointer`. I think this would be more natural as:

```
  oldErr := unsafe.Pointer(reply.Header().Error)
  newErr := &proto.Error{Message: "boom"}
  atomic.SwapPointer(&oldErr, unsafe.Pointer(newErr))
```